docs/secret-service.md

```diff
@@ -0,0 +1,153 @@
+than the standard QUIC one, though this can have security implications if not
+implemented correctly. The QUIC implementation is fast and secure using TLS 1.3
+already.
+
+TLS configuration is handled via a `rustls` server config.
+
+#### `secret-service-client`
+
+[`tokio`](https://tokio.rs/) & [`quinn`](https://github.com/quinn-rs/quinn)-based implementation of an S2 client. Uses the QUIC request/response pattern that standard S2 servers expect.
+
+If you swap your transport protocol, you'd have to create a new client too. You could
+also use this as a template for swapping to another event loop like [`monoio`](https://github.com/bytedance/monoio) or
+QUIC implementation.
+
+TLS configuration is handled via a `rustls` client config.
+
+### Available methods
+
+S2's API is made up of several stubs of related functionality. These are implemented as traits in `secret-service-proto::v2`.
+
+| Stub                  |                                                                                                                                                    |
+| --------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Musig2Signer`        | Operations around Musig2 signing sessions. Includes the various operations where secrets are involved. All other logic is delegated to the client. |
+| `P2PSigner`           | BIP340 Schnorr signer interface for signing operator's messages for the P2P network                                                                |
+| `StakeChainPreimages` | Deterministic creation of preimages for the stakechain that the operator uses to fulfil withdrawals.                                               |
+| `WotsSigner`          | Winternitz One-Time Signatures are used as a hash-based primitive to transfer state across Bitcoin UTXOs without native support.                   |
+
+The exact operations available along with usage and implementation specification
+can be viewed in the Rust docs.
+
+### Networking
+
+Communication between client and server happens over a custom QUIC-based
+binary protocol.
+
+The client is responsible for keeping the connection open with QUIC keep-alive
+messages.
+
+#### Wire protocol
+
+The wire protocol is a request/response pattern, somewhat RPC-like. For each
+request, a client opens a new bidirectional QUIC stream. It then encodes a request,
+which is a `VersionedClientMessage`. This contains any inputs/parameters to the
+function.
+
+For each connection, the server will wait for new incoming streams. It will accept
+these and then attempt to read & parse the client's request. The request is then
+executed, and a `VersionedServerMessage` is sent back over the stream to the client.
+The stream is then closed on both sides.
+
+Messages are encoded via [`rkyv`](https://github.com/rkyv/rkyv), which is a Rust exclusive, high performance
```

Comment:
Does rkyv commit to a stable format? I usually don't like serialization schemes being delegated outside of the codebase because it means you don't have control over the api stability.